doc/source/whatsnew/v1.6.0.rst

```diff
@@ -175,6 +175,7 @@ Missing
 
 MultiIndex
 ^^^^^^^^^^
+- Bug in :meth:`MultiIndex.difference` losing extension array dtype (:issue:`48606`)
```

Comment:
Could mention the performance improvement too?

Reply:
Was only a bit, but could add it anyway if you prefer?

Reply:
Might as well :)

Reply:
Added